### PR TITLE
Raghav/initial commit

### PR DESCRIPTION
**Pull request by rakatyal**
case 1. Add a new type definition.
- [ ] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [ ] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [ ] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url http://api.jquery.com/html .
  - it has been reviewed by a DefinitelyTyped member.

### CODE REVIEW

File: base-x/index.d.ts [hunk 1]
```diff
@@ -3,6 +3,10 @@
 // Definitions by: Ilya Mochalov <https://github.com/chrootsu>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
+declare var base: BaseX.Base;
+export = base;
+export as namespace BaseX;
```
Comment (RyanCavanaugh) [inline base-x/index.d.ts]: This library doesn't appear to be UMD

File: commander/commander-tests.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // NOTE: import statement can not use in TypeScript 1.0.1
-var program:commander.IExportedCommand = require('commander');
+var program = require('commander');
```
Comment (RyanCavanaugh) [inline commander/commander-tests.ts]: Should be `import program = require('commander');` (not `var`)

File: commonmark/index.d.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@
 // Definitions by: Nico Jansen <https://github.com/nicojs>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
+export = commonmark;
```
Comment (RyanCavanaugh) [inline commonmark/index.d.ts:6]: This library appears to be UMD

Comment (rakatyal) [inline commonmark/index.d.ts:6]: Yup. Missed it.

File: cropperjs/index.d.ts [hunk 4]
```diff
@@ -4,6 +4,10 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 
+declare const Cropper: typeof cropperjs.Cropper;
+export = Cropper;
+export as namespace cropperjs;
```
Comment (RyanCavanaugh) [inline cropperjs/index.d.ts]: This module exposes itself from the global `Cropper`, so this should be `export as namespace Cropper`

File: d3-dsv/index.d.ts [hunk 5]
```diff
@@ -4,18 +4,16 @@
...
-        encoding?: string) => _d3dsv.D3Dsv;
-    export = loader;
-}
+declare var loader: (
+    /** the symbol used to seperate cells in the row.*/
+    delimiter: string,
+    /** example: "text/plain" */
+    encoding?: string) => _d3dsv.D3Dsv;
+export = loader;
+export as namespace _d3dsv;
```
Comment (RyanCavanaugh) [inline d3-dsv/index.d.ts]: This is exposed from `d3_dsv`

File: diff/index.d.ts [hunk 6]
```diff
@@ -3,6 +3,8 @@
 // Definitions by: vvakame <https://github.com/vvakame/>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
+export = JsDiff;
```
Comment (RyanCavanaugh) [inline diff/index.d.ts:6]: UMD module `JsDiff`

File: domready/index.d.ts [hunk 7]
```diff
@@ -5,6 +5,4 @@
 
 declare function domready(callback: () => any) : void;
```
Comment (RyanCavanaugh) [inline domready/index.d.ts:6]: UMD under `domready`

File: donna/donna-tests.ts [hunk 8]
```diff
@@ -1,7 +1,7 @@
 
 
 import donna = require("donna");
-var metadata = donna.generateMetadata(['/path/to/my-module', '/path/to/another-module']);
+var metadata = donna(['/path/to/my-module', '/path/to/another-module']);
```
Comment (RyanCavanaugh) [inline donna/donna-tests.ts]: :warning: Don't make test changes like this without consulting the docs for the library.

https://github.com/atom/donna#from-your-code

File: donna/index.d.ts [hunk 9]
```diff
@@ -3,9 +3,8 @@
 // Definitions by: vvakame <https://github.com/vvakame/>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-declare module "donna" {
-    function generateMetadata(modules: string[]): DonnaTypes.Metadata;
-}
+declare function generateMetadata(modules: string[]): DonnaTypes.Metadata;
+export = generateMetadata;
```
Comment (RyanCavanaugh) [inline donna/index.d.ts]: This should be `export function generateMetadata ...` with no `export =` declaration

File: simple-mock/simple-mock-tests.ts [hunk 10]
```diff
@@ -15,7 +15,7 @@ import Bluebird = require('bluebird');
 describe('simple', function () {
   describe('spy()', function () {
     describe('for noop function', function () {
-      let spyFn: Simple.Spy<void>;
```
Comment (rakatyal) [inline simple-mock/simple-mock-tests.ts:18]: Modified this module in a better way in later commit to keep the 'casts' in tests.

**Pull request merged**